src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +789,234 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * This overrides the add_data method from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
+			$args = $this->get_normalized_script_args( $handle, $value );
+			if ( $args['in_footer'] ) {
+				parent::add_data( $handle, 'group', 1 );
+			}
+			return parent::add_data( $handle, $key, $args );
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script() {
+		foreach ( $this->registered as $handle => $script ) {
+			// Non standalone scripts in the after position, of type async or defer, are usually delayed.
+			$strategy = $this->get_intended_strategy( $handle );
+			if ( $this->is_non_blocking_strategy( $strategy ) &&
+				$this->has_non_standalone_inline_script( $handle, 'after' )
+			) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Normalize the data inside the $args parameter and support backward compatibility.
+	 *
+	 * @param string        $handle Name of the script.
+	 * @param array         $args     {
+	 *      Optional. Additional script arguments. Default empty array.
+	 *
+	 *      @type boolean   $in_footer    Optional. Default false.
+	 *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+	 * }
+	 * @return array        Normalized $args array.
+	 */
+	private function get_normalized_script_args( $handle, $args = array() ) {
+		$default_args = array(
+			'in_footer' => false,
+			'strategy'  => 'blocking',
+		);
+
+		// Handle backward compatibility for $in_footer.
+		if ( true === $args ) {
+			$args = array( 'in_footer' => true );
+		}
+
+		return wp_parse_args( $args, $default_args );
+	}
+
+	/**
+	 * Get all dependents of a script.
+	 *
+	 * @param string $handle The script handle.
+	 * @return array Array of script handles.
+	 */
+	private function get_dependents( $handle ) {
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding dependents of the script passed to this method.
+		foreach ( $this->registered as $registered_handle => $args ) {
```

Comment:
this loop execution time will grow as the number of registered handles grows. Probably fine, but I do have a concern that on large sites with many scripts it could become a bit heavy. a potential optimization here would be building a map of handle->array(dependents) the first time it is called, then use the map for subsequent calls.

Reply:
@adamsilverstein I've amended this in a recent commit to make sue of a private var to maintain the mapping, and use this for a given handle should it exist.